File: LAB_3/LAB_3_Java/src/Consumer.java
public class Consumer implements Runnable
{
    private final int _maxItem;
    private final Storage _storage;

    public Consumer(int maxItem, Storage storage)
    {
        _maxItem = maxItem;
        _storage = storage;
    }

    @Override
    public void run()
    {
        for (int i = 0; i < _maxItem; i++)
        {
            String item;
            try
            {
                _storage.acquireEmpty();
                Thread.sleep(1000);
                _storage.acquireAccess();

                item = _storage.getItem();
                _storage.removeItem();
                System.out.println("Took " + item);

                _storage.releaseAccess();
                _storage.releaseFull();

            }
            catch (InterruptedException e)
            {
                e.printStackTrace();
            }
        }
    }
}

File: LAB_3/LAB_3_Csharp/LAB_3_Csharp/Consumer.cs
using System;
using System.Threading;

namespace LAB_3_Csharp
{
    internal class Consumer
    {
        private int _maxItem = 0;
        private Storage _storage;
        public Consumer(int maxItem, Storage storage)
        {
            _maxItem = maxItem;
            _storage = storage;
        }

        public void TakeItem()
        {
            for (int i = 0; i < _maxItem; i++)
            {
                _storage.AcquireEmpty();
                Thread.Sleep(1000);
                _storage.AcquireAccess();

                string item = _storage.GetItem();
                _storage.RemoveItem();

                _storage.ReleaseFull();

                _storage.ReleaseAccess();

                Console.WriteLine("Took " + item);
            }
        }
    }
}
